modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestBodyStream.java

```diff
@@ -28,7 +28,7 @@ public class Netty4HttpRequestBodyStream implements HttpBody.Stream {
 
     private final Channel channel;
     private final Queue<HttpContent> chunkQueue = new ArrayDeque<>();
-    private boolean requested = false;
+    private boolean requested = true;
```

Comment:
This breaks stream interface. First chunk will be passed to handler before `.next()` invocation, also will break my current tests. My intention was that you call `.next()` on `RestChannelConsumer.accept()`.

Reply:
Maybe also make it memory aware.

Reply:
Yes sorry. I meant to ask you how you intended the first chunk to be sent as tests were hanging at first.

I will make this change in accept.